### PR TITLE
Add section about comparing changes to the git bootcamp

### LINKED ISSUE #794: Add git diff tricks to git cheat sheet section
After python/cpython#30745 we can do some neat git diff tricks:

    $ git switch <somebranch-with-autogenerated-changes>
    $ git diff main ':(attr:!generated)'
    $ git config diff.generated.binary true
    $ git diff main

It would be nice to document these in the devguide.

### PR DESCRIPTION
**Pull request by erlend-aasland**
Fixes #794

### CODE REVIEW

File: gitbootcamp.rst [hunk 1]
```diff
@@ -240,6 +240,27 @@ Commit the files:
...
+   git diff main
+
+Exclude generated files from diff (note the single quotes)::
+
+   git diff main ':(attr:!generated)'
+
+Exclude generated files from diff by default::
+
+   git config diff.generated.binary true
+
```
Comment (zware) [inline gitbootcamp.rst:263]: Should we mention that both of these options are driven by the `generated` attribute in `.gitattributes`?

Comment (erlend-aasland) [reply gitbootcamp.rst:264]: Sounds good. PTAL.

File: gitbootcamp.rst [hunk 2]
```diff
@@ -253,14 +253,21 @@ Compare to the ``main`` branch::
...
+Exclude generated files from diff using an `attr <_gitattributes>`_
+`pathspec <_pathspec>`_ (note the single quotes)::
 
    git diff main ':(attr:!generated)'
 
 Exclude generated files from diff by default::
 
    git config diff.generated.binary true
 
+The `generated` `attribute <_gitattributes>`_ is defined in `.gitattributes`,
```
Comment (zware) [inline gitbootcamp.rst]: reST doesn't like single backticks when it's not a role or link:
```suggestion
The ``generated`` `attribute <_gitattributes>`_ is defined in ``.gitattributes``,
```

I also don't think we need to repeat the link to the gitattributes docs:

```suggestion
The ``generated`` attribute is defined in ``.gitattributes``,
```

File: gitbootcamp.rst [hunk 3]
```diff
@@ -240,6 +240,32 @@ Commit the files:
...
+single quotes)::
+
+   git diff main ':(attr:!generated)'
+
+Exclude generated files from diff by default::
+
+   git config diff.generated.binary true
+
+The ``generated`` `attribute <https://git-scm.com/docs/gitattributes>`_ is
+defined in `.gitattributes`, found in the repository root.
```
Comment (zware) [inline gitbootcamp.rst:267]:
```suggestion
defined in :file:`.gitattributes`, found in the repository root.
```

Comment (erlend-aasland) [reply gitbootcamp.rst:267]: Oh, I didn't know about that role. Thanks.

**Comment by erlend-aasland** [comment]
Thanks for the improvements, Zachary!

**Pull request merged**